core/trino-main/src/test/java/io/trino/sql/gen/TestExpressionCompiler.java

```diff
@@ -1476,6 +1483,27 @@ public void testFunctionCallRegexp()
         Futures.allAsList(futures).get();
     }
 
+    @Test
+    public void testSearchInterruptible()
+            throws Exception
+    {
+        String pattern = "\\((.*,)+(.*\\))";
+        String source = Files.readString(Paths.get(EXTRA_LONG_SOURCE_PATH));
+        RegularExpressionRunnable regularExpressionRunnable = new RegularExpressionRunnable(source, pattern);
+        Thread searchChildThread = new Thread(regularExpressionRunnable);
+
+        searchChildThread.start();
+
+        // wait for the child thread to make some progress
+        searchChildThread.join(1000);
```

Comment:
this seems like reasonable wait to make sure it has indeed started executing. however we could examine thread stacktrace until it's indeed stuck. i'm on the fence if it'd be too defensive.  @findepi @losipiuk wdyt from similar tests?